spec/javascripts/school_administrator_dashboard/students_table_spec.jsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import StudentsTable from '../../../app/assets/javascripts/school_administrator_dashboard/dashboard_components/students_table.jsx';
+import * as Data from './DashboardTestData.js';
+
+describe('Dashboard Students Table', () => {
+  const table = shallow(<StudentsTable rows={Data.Students}/>);
+
+  it('renders the students list', () => {
+    expect(table.find("div").hasClass("StudentsList")).toEqual(true);
+  });
+
+  it('tallies the total events', () => {
+    expect(table.instance().totalEvents()).toEqual(12);
+  });
+
+  it('orders the students by total events by default', () => {
+    expect(table.instance().orderedRows()[5].first_name).toEqual("Arlecchino");
+  });
+
+  it('orders students by name when Name is clicked', () => {
+    table.find('.sort-header').first().simulate('click');
+    expect(table.state().sortBy).toEqual("last_name");
```

Comment:
Would it make sense to have an expectation here about the first value in the table changing?

Reply:
It would, @alexsoble. And it's a good test b/c it may have uncovered a bug in the sorthelpers. I'll check on what's going on there and update this then.

Reply:
@alexsoble So looking into this a bit, the function I'm using to sort here is the same used by the school overview. It sorts by last names, but doesn't do a secondary sort on first names. That might be a change worth making at some point, but since I don't know why it was implemented the way it was, I'm inclined to leave the behavior as is and alter the test data so secondary sorts aren't a problem.

Reply:
@edavidsonsawyer Agreed!